src/herbivoresAndCarnivores.js

```diff
@@ -1,17 +1,47 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  // static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    Animal.alive.push(this);
+  }
+
+  bite(animal) {
+    if (!((animal instanceof Carnivore) || (animal.hidden === true))) {
```

Comment:
```suggestion
    if (animal instanceof Herbivore && !animal.hidden) {
```

and remove else block